$MirOS: ports/net/djbdns/patches/patch-src_roots_c,v 1.2 2005/12/14 23:28:38 tg Exp $
--- src/roots.c.orig	Sun Feb 11 21:11:23 2001
+++ src/roots.c	Sat May 30 21:47:38 2009
@@ -6,6 +6,9 @@
 #include "error.h"
 #include "direntry.h"
 #include "ip4.h"
+#ifdef DJB_V6ONLY
+#include "ip6.h"
+#endif /* DJB_V6ONLY */
 #include "dns.h"
 #include "openreadclose.h"
 #include "roots.h"
@@ -22,7 +25,11 @@ static int roots_find(char *q)
     j = dns_domain_length(data.s + i);
     if (dns_domain_equal(data.s + i,q)) return i + j;
     i += j;
+#ifndef DJB_V6ONLY
     i += 64;
+#else
+    i += 512;
+#endif /* DJB_V6ONLY */
   }
   return -1;
 }
@@ -40,12 +47,20 @@ static int roots_search(char *q)
   }
 }
 
+#ifndef DJB_V6ONLY
 int roots(char servers[64],char *q)
+#else
+int roots(char servers[512],char *q)
+#endif /* DJB_V6ONLY */
 {
   int r;
   r = roots_find(q);
   if (r == -1) return 0;
+#ifndef DJB_V6ONLY
   byte_copy(servers,64,data.s + r);
+#else
+  byte_copy(servers,512,data.s + r);
+#endif /* DJB_V6ONLY */
   return 1;
 }
 
@@ -60,7 +75,11 @@ static int init2(DIR *dir)
   const char *fqdn;
   static char *q;
   static stralloc text;
+#ifndef DJB_V6ONLY
   char servers[64];
+#else
+  char servers[512];
+#endif /* DJB_V6ONLY */
   int serverslen;
   int i;
   int j;
@@ -85,15 +104,29 @@ static int init2(DIR *dir)
       j = 0;
       for (i = 0;i < text.len;++i)
 	if (text.s[i] == '\n') {
+#ifndef DJB_V6ONLY
 	  if (serverslen <= 60)
 	    if (ip4_scan(text.s + j,servers + serverslen))
 	      serverslen += 4;
+#else
+	  if (serverslen <= 512 - 16)
+	    if (ip6_scan(text.s + j,servers + serverslen))
+	      serverslen += 16;
+#endif /* DJB_V6ONLY */
 	  j = i + 1;
 	}
+#ifndef DJB_V6ONLY
       byte_zero(servers + serverslen,64 - serverslen);
+#else
+      byte_zero(servers + serverslen,512 - serverslen);
+#endif /* DJB_V6ONLY */
 
       if (!stralloc_catb(&data,q,dns_domain_length(q))) return 0;
+#ifndef DJB_V6ONLY
       if (!stralloc_catb(&data,servers,64)) return 0;
+#else
+      if (!stralloc_catb(&data,servers,512)) return 0;
+#endif /* DJB_V6ONLY */
     }
   }
 }
